Ginger/GingerCoreNET/Database/DatabaseOperations.cs

```diff
@@ -492,18 +493,30 @@ public List<string> GetTablesList(string Keyspace = null)
                         GingerCosmos objGingerCosmos = new GingerCosmos();
                         Database.ConnectionString = GetConnectionString();
                         objGingerCosmos.Db = Database;
-                        rc = objGingerCosmos.GetTableList(Keyspace);
+                        rc =  objGingerCosmos.GetTableList(Keyspace);
+                    }
+                    else if (Database.DBType == eDBTypes.Oracle)
+                    {
+                        string[] restr = new string[1];
+                        restr[0] = GetConnectedUsername();
+                        DataTable table = await oConn.GetSchemaAsync("Tables", restr);
+                        string tableName = "";
+                        foreach (DataRow row in table.Rows)
+                        {
+                            tableName = (string)row[1];
+                             rc.Add(tableName);
+                        }
                     }
                     else
                     {
-                        DataTable table = oConn.GetSchema("Tables");
+                        DataTable table = await Task.Run(() => oConn.GetSchema("Tables"));
                         string tableName = "";
                         foreach (DataRow row in table.Rows)
                         {
                             switch (Database.DBType)
                             {
```

Comment:
Remove oracle case from switch as it is handled above
Also check if switch case is needed here as its being evealuated for each row in table.rows.